modules/argocd-repo/main.tf

```diff
@@ -1,14 +1,17 @@
 locals {
   enabled = module.this.enabled
+
   environments = local.enabled ? {
     for env in var.environments :
     (format(
-      "${env.tenant != null ? "%[1]s/" : ""}%[2]s-%[3]s",
+      "${env.tenant != null ? "%[1]s/" : ""}%[2]s-%[3]s${length(env.attributes) > 0 ? "-%[4]s" : "%[4]s"}",
```

Comment:
Because the default option attributes is `[]` and the `format()` function takes 4 inputs.


If we use empty string instead, then Terraform `format()` will fail:
```
│   on main.tf line 11, in locals:
│    6:     (format(
│    7:       "${env.tenant != null ? "%[1]s/" : ""}%[2]s-%[3]s${length(env.attributes) > 0 ? "-%[4]s" : ""}",
│    8:       env.tenant,
│    9:       env.environment,
│   10:       env.stage,
│   11:       "${join("-", env.attributes)}"
│   12:     )) => env
│     ├────────────────
│     │ env.attributes is empty list of string
│
│ Invalid value for "args" parameter: too many arguments; only 3 used by format string.
```